mmfunctions/anomaly.py

```diff
@@ -1127,78 +1134,33 @@ class MatrixProfileAnomalyScore(BaseTransformer):
         ERROR_SCORES = 1e-16
 
         def __init__(self, input_item, window_size, output_item):
-            super().__init__()
+            super().__init__(input_item, window_size, [output_item])
             logger.debug(f'Input item: {input_item}')
-            self.input_item = input_item
-            self.window_size = window_size
-            self.output_item = output_item
-            self.whoami = 'MatrixProfile'
-
-        def prepare_data(self, df_entity):
 
-            logger.debug(self.whoami + ': prepare Data')
+            self.whoami = 'MatrixProfile'
 
-            # operate on simple timestamp index
-            if len(df_entity.index.names) > 1:
-                index_names = df_entity.index.names
-                dfe = df_entity.reset_index(index_names[1:])
-            else:
-                dfe = df_entity
 
-            # interpolate gaps - data imputation
-            try:
-                dfe = dfe.interpolate(method="time")
-            except Exception as e:
-                logger.error('Prepare data error: ' + str(e))
+        def score(self, temperature):
 
-            # one dimensional time series
-            analysis_input = dfe[self.input_item].fillna(0).to_numpy(dtype=np.float64)
+            scores = []
 
-            return dfe, analysis_input
+            for output_item in self.output_items:
```

Comment:
Optional command to remove for loop (%%timeit analysis: a few nanoseconds better when length of self.output_items is 1; considerable improvement (for-loop takes micro sec compared to nanosecond) when the length increases)
```python3
np.zeros((len(self.output_items), ) + temperature.shape)
```

Reply:
Thanks, never thought of adding tuples